modules/scylladb/scylladb.go

```diff
@@ -0,0 +1,142 @@
+package scylladb
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/docker/go-connections/nat"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	port           = nat.Port("9042/tcp")
+	shardAwarePort = nat.Port("19042/tcp")
+)
+
+type ScyllaDBContainer struct {
+	testcontainers.Container
+}
+
+// WithConfigFile sets the YAML config file to be used for the cassandra container
+// It will also set the "configFile" parameter to the path of the config file
+// as a command line argument to the container.
+func WithConfigFile(configFile string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		cf := testcontainers.ContainerFile{
+			HostFilePath:      configFile,
+			ContainerFilePath: "/etc/scylla/scylla.yaml",
+			FileMode:          0o755,
+		}
+		req.Files = append(req.Files, cf)
+
+		return nil
+	}
+}
+
+// WithShardAwareness enable shard-awareness in the ScyllaDB container so you can use the `19042` port.
+func WithShardAwareness() testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		req.ExposedPorts = append(req.ExposedPorts, shardAwarePort.Port())
+		req.WaitingFor = wait.ForAll(req.WaitingFor, wait.ForListeningPort(shardAwarePort))
+		return nil
+	}
+}
+
+// WithAlternator enables the Alternator (DynamoDB Compatible API) service in the ScyllaDB container.
+// It will set the "alternator-port" parameter to the specified port.
+// It will also set the "alternator-write-isolation" parameter to "always" as a command line argument to the container.
+func WithAlternator(alternatorPort uint) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		setCommandFlag(req, "--alternator-port", strconv.Itoa(int(alternatorPort)))
```

Comment:
you have been changing `int` to `string` 3 times.
Extract it once in a variable then use it where it's needed